js/collaborator-agent/src/capabilities/template/template.ts

```diff
@@ -33,91 +18,58 @@ export interface TemplateResult {
  */
-export class TemplateCapability {
-    constructor() {
-        // Initialize any resources your capability needs
-    }
+export class TemplateCapability extends BaseCapability {
+  readonly name = 'template';
 
-    /**
-     * Process a user request within this capability's domain
-     * @param context - The message context containing user info, conversation details, etc.
-     * @param options - Additional options like storage, time ranges, etc.
-     * @returns Promise<TemplateResult> - The result of processing the request
-     */
-    async processRequest(context: MessageContext, options: TemplateOptions = {}): Promise<TemplateResult> {
-        try {
-            console.log(`🔧 Template Capability processing request: ${context.text}`);
+  createPrompt(context: MessageContext): ChatPrompt {
+    const templateModelConfig = this.getModelConfig('template');
 
-            // Extract relevant options
-            const {
-                storage,
-                citationsArray,
-                calculatedStartTime,
-                calculatedEndTime,
-                timespanDescription
-            } = options;
+    const prompt = new ChatPrompt({
+      instructions: TEMPLATE_PROMPT,
+      model: new OpenAIChatModel({
+        model: templateModelConfig.model,
+        apiKey: templateModelConfig.apiKey,
+        endpoint: templateModelConfig.endpoint,
+        apiVersion: templateModelConfig.apiVersion,
+      }),
+    }).function('process_template_request', 'Process a template-specific request', TEMPLATE_FUNCTION_SCHEMA,
+      async (args: TemplateFunctionArgs) => {
 
-            // TODO: Implement your capability's core logic here
-            
-            // Example: Basic request processing
-            const userRequest = context.text;
-            // const conversationId = context.conversationId;
-            // const userId = context.userId;
-            const userName = context.userName;
-            const isPersonalChat = context.isPersonalChat;
-            // const members = context.members; // Available conversation members
+        this.logger.debug(`🔧 Template Capability processing request with args:`, args);
 
-            // Example: Use time range if provided
-            if (calculatedStartTime && calculatedEndTime) {
-                console.log(`📅 Processing with time range: ${timespanDescription}`);
-                // Handle time-based queries
-            }
+        // TODO: Implement your capability's core logic here
+        
+        // Example: Basic request processing
+        const userRequest = context.text;
+        const userName = context.userName;
+        const isPersonalChat = context.isPersonalChat;
 
-            // Example: Use storage if provided
-            if (storage) {
-                // Interact with storage
-                // const data = await storage.get(conversationId, 'some-key');
-            }
+        // TODO: Replace this with your actual processing logic
```

Comment:
This is great. Make sure you have a section on how to add a new capability (if you don't already)